File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0;
        int j = height.length - 1;
        int max_area = 0;
        while(i < j){
            int current_base = j - i;
            int min_side = Math.min(height[i],height[j]);
            int current_area = current_base * min_side;

            if(current_area > max_area){
                max_area = current_area;
            }
            if(height[i] >= height[j]) j--;
            else i++;
        }
        return max_area;
    }
}

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxwater = 0;
        int n = height.size();
        int lp = 0;
        int rp = n - 1;
        while(lp < rp){
            int width = rp - lp;
            int heigh = min(height[lp],height[rp]);
            int currwater = width * heigh;
            maxwater = max(maxwater,currwater);
            if(height[lp] < height[rp]){
                lp++;
            }
            else{
                rp--;
            }
        }
        return maxwater;
    }
};